### ISSUE TITLE
PHP 7.4 support

### ISSUE THREAD

**Issue by remicollet** [protocolbuffers/protobuf#6180]
For now, build fails

```
/dev/shm/BUILD/php74-php-pecl-protobuf-3.8.0~RC1/NTS/storage.c: In function 'native_slot_set':
/dev/shm/BUILD/php74-php-pecl-protobuf-3.8.0~RC1/NTS/storage.c:106:9: error: too few arguments to function 'zend_assign_to_variable'
         zend_assign_to_variable(cached_zval, value, IS_CV);
         ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/remi/php74/root/usr/include/php/Zend/zend_API.h:29,
                 from /opt/remi/php74/root/usr/include/php/main/php.h:37,
                 from ./protobuf.h:34,
                 from /dev/shm/BUILD/php74-php-pecl-protobuf-3.8.0~RC1/NTS/storage.c:32:
/opt/remi/php74/root/usr/include/php/Zend/zend_execute.h:81:33: note: declared here
 static zend_always_inline zval* zend_assign_to_variable(zval *variable_ptr, zval *value, zend_uchar value_type, zend_bool strict)
                                 ^~~~~~~~~~~~~~~~~~~~~~~

```

**Comment by TeBoring** [comment]
@remicollet trying to install php 7.4 from source.
However, I met the error:
```
/php-src/ext/json/json_parser.tab.c: No such file or directory
```
Seems the json_parser.tab.c is needed for json extension, but it was never generated in Makefile.
https://github.com/php/php-src/blob/php-7.4.0/ext/json/config.m4

**Comment by remicollet** [comment]
Use the official tarballs, or you have some additional steps (buildconf, ...)

**Comment by TeBoring** [comment]
Even with buildconf, building from src still doesn't work.
I think the config.m4 is missing something like https://github.com/php/php-src/blob/php-7.4.0/ext/json/config.w32#L12

**Comment by remicollet** [comment]
Sorry, but I don't even know what Windows is

**Comment by cmb69** [comment]
I don't think this about Windows (the PECL package does not even have config.w32).

**Comment by DRoet** [comment]
Would love to see this, FYI: [PHP 7.4 has been released](https://www.php.net/index.php#id2019-11-28-1)

**Comment by camry** [comment]
I also encountered the same problem.

**Comment by DRoet** [comment]
this can be closed now that `v3.11.2` is out
